File: java/template/agent-infrastructure/src/main/java/com/workshop/agent/infrastructure/RepeatingLanguageModel.java
package com.workshop.agent.infrastructure;

import com.workshop.agent.application.LanguageModel;
import com.workshop.agent.application.Message;

import java.util.List;

public class RepeatingLanguageModel implements LanguageModel {
    @Override
    public Message prompt(List<Message> messages) {
        Message lastMessage = messages.get(messages.size() - 1);
        return new Message("assistant", "You said: \"" + lastMessage.content() + "\"");
    }
}

File: csharp/template/Agent/Agent.Infrastructure/RepeatingLanguageModel.cs
using Agent.Application;

namespace Agent.Infrastructure;

public class RepeatingLanguageModel : ILanguageModel
{
    public Message Prompt(IEnumerable<Message> messages)
    {
        var list = messages.ToList();
        return new Message("assistant", "You said: \"" + list.Last().Content + "\"");
    }
}
